pkg/azuredx/datasource.go

```diff
@@ -35,7 +36,7 @@ func NewDatasource(settings backend.DataSourceInstanceSettings) (instancemgmt.In
 	}
 	adx.settings = datasourceSettings
 
-	tokenProvider := tokenprovider.NewAccessTokenProvider(tokenCache, datasourceSettings.ClientID, datasourceSettings.TenantID, "AzurePublic", datasourceSettings.Secret, []string{"https://kusto.kusto.windows.net/.default"})
+	tokenProvider := tokenprovider.NewAccessTokenProvider(tokenCache, datasourceSettings.ClientID, datasourceSettings.TenantID, datasourceSettings.AzureCloud, datasourceSettings.Secret, []string{"https://kusto.kusto.windows.net/.default"})
```

Comment:
I'm not sure I see if a default value being set for the cloud in the backend?

`resolveAuthorityHost` has a fallback value, but I wonder if we should set a default up here at the top so elsewhere in the code we know that the value is always set (like for the [cache key](https://github.com/grafana/azure-data-explorer-datasource/blob/074ff5cc873f98e6713e7cb873fdd242fa0537f5/pkg/azuredx/tokenprovider/token_provider.go#L78)?)

Reply:
I actually think it's sufficient to handle this is in the token provider. The tokenprovider package is meant to become the azure-sdk-go, and I'd like to push guards like this into the SDK to protect from future plugin developers from making mistakes. WDYT?

Reply:
Yeah that makes sense. Just as long as value is consistent.